shared/bash_remediation_functions/lineinfile.sh

```diff
@@ -0,0 +1,106 @@
+# Functions which minic Ansible's lineinfile module.
+# dependencies: die.sh, variables.sh
+
+function lineinfile() {
+    local path="$1"
+    local regex="$2"
+    local new_line="$3"
+    local create="$(normalize_boolean "$4" "false")"
+    local state="$(normalize_state "$5" "present")"
+    local insert_after="$6"
+    local insert_before="$7"
+    local insensitive="$(normalize_boolean "$8" "true")"
+
+    [ ! -e "$path" ] && [ "$create" != "true" ] && die "Path '$path' wasn't found on this system and not creating. Refusing to continue."
+    [ ! -e "$path" ] && [ "$create" == "true" ] && touch "$path"
+
+    if [ "$state" == "absent" ]; then
+        [ -n "$new_line" ] && echo "Ignoring parameter new_line in absent mode..."
+        echo "Saving $path to $path.bak and removing line defined by the regex '$regex'..."
+        if [ "$insensitive" == "true" ]; then
+            LC_ALL=C sed --in-place=.bak "/$regex/Id" "$path"
+        else
+            LC_ALL=C sed --in-place=.bak "/$regex/d" "$path"
+        fi
+    elif [ "$state" == "present" ]; then
+        local grep_args=("-q" "-m" "1")
+        if [ "$insensitive" == "true" ]; then
+            grep_args+=("-i")
+        fi
+        if [ -z "$regex" ] || ! LC_ALL=C grep "${grep_args[@]}" "$regex" "$path"; then
+            if [ -z "$insert_after" ] && [ -z "$insert_before" ] || [ "$insert_after" == "EOF" ]; then
+                # Insert at the end of the file
+                cp "$path" "$path.bak"
+                echo "$new_line" >> "$path"
+            elif [ "$insert_before" == "BOF" ]; then
+                # Insert at the beginning of the file
+                cp "$path" "$path.bak"
+                echo "$new_line" > "$path"
```

Comment:
This is a nitpick, but `printf '%s\n' "$new_line"` is [slightly better](https://unix.stackexchange.com/a/65819/260749).

Reply:
Ah right, we had this discussion last time I went modifying bash... :-) I'll change this when I update the PR.